---
title: "Experimental Designs"
author: "Brian S. Yandell, UW-Madison, github.com/byandell/PlantSysGen"
date: "January 2017"
output:
  ioslides_presentation: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Outline

- common breeding designs
    + backcross (BC)
    + intercross (F2)
    + doubled haploid (DH)
- advanced intercross lines
    + recombinant inbred lines (RILs)
    + near isogenic lines (NILs) & consomics
    + multi-parent populations (MPP)

## common breeding designs

<img src="../Figs/backcross_bw.jpg" 
  alt="Backcross" 
  style="width: 350px; float:right"/>

- 2 (inbred) founder alleles
- 2 generations
- backcross (BC): 1 meiosis
- doubled haploid (DH): 1 meiosis
- intercross (F2): 2 meioses

<img src="../Figs/intercross_bw.jpg" 
  alt="Intercross" 
  style="width: 350px; float:left"/>

<img src="../Figs/DH_bw.jpg" 
  alt="Intercross" 
  style="width: 350px; float:right"/>

## recombinant inbred lines (RIL)

<img src="../Figs/nrg2291-f4.jpeg" 
  alt="RIL" 
  style="width: 300px; float:right"/>

- 2 or more inbred founders
- single F1 self-pollinated
- generations of random mating
- generations of selfing
- aim for homozygosity at all loci

<span style="font-size: 50%">www.nature.com/nrg/journal/v9/n3/images/nrg2291-f4.jpg</span>

## Selfing vs sib mating

![](../Figs/rigenome_fig2_lg.png)

<a href="http://kbroman.org/pages/publpics/rigenome_fig2.html"><span style="font-size: 75%">Broman (2005) Genetics</span></a>

## near isogenic lines (NIL)

<img src="../Figs/NIILpanel_Univ-of-Illinois-Chicago_w.jpeg" 
  alt="RIL" 
  style="width: 300px; float:left"/>

<img src="../Figs/meddic.jp.png" 
  alt="RIL" 
  style="width: 450px; float:right"/>


<span style="font-size: 50%">Rebecca Nelson blog.generationcp.org/category/women-in-science-2/</span>

<span style="font-size: 50%">meddic.jp/isogenic_line</span>

## Advanced Intercrosses

<img src="../Figs/physiolgenomics.jpg" 
  alt="AI_Woods" 
  style="width: 800px"/>

<span style="font-size: 75%">Leah Solberg Woods doi:10.1152/physiolgenomics.00127.2013</span> 

## multi-parent populations

<img src="../Figs/DO_laura_vanderploeg.jpg" 
  alt="RIL" 
  style="width: 300px; float:right"/>

- more than 2 inbred parents (4,8,20)
- developed over generations
    + generations of cross-breeding
    + generations of selfing (or sibs)
- increased meiotic events
    + fine mapping to small region
    + SNP level in one generation

<span style="font-size: 75%">Laura Vanderploeg, Jackson Labs</span>

## Osborn _Brassica napus_ intercross

- 104 doubled haploid (DH) lines
- 300 markers on 19 chromosomes
    + originally scattered linkage groups
- 9 phenotypes (flowering time & seedling survival)

<span style="font-size: 75%">Ferreira, Satagopan, Yandell, Williams, Osborn (1995) TAG</span>  
<span style="font-size: 75%">Satagopan, Yandell, Newton, Osborn (1996) Genetics</span>

## Moore/Spalding _A. thaliana_ NIL & RILs

- _Arabidopsis thaliana_ Ler x Cvi population
    + 92 near-isogenic lines (NIL); 2525 seedlings
    + 162 RILs; 2132 (RIL1) or 2325 (RIL2) seedlings
- genotypes: 102 (NIL) or 234 (RILs) markers on 5 chr
- phenotypes: 241 root tip angles, every 2 min
- automated image acquisition & analysis
    + images: 7000 lines x 241 time points
    + genome scans across all time points
- botany / computer science / biostatistics collaboration

<span style="font-size: 75%">Moore, Johnson, Kwak, Livny, Broman, Spalding (2013)</span>

## Diversity Outbred example

<img src="../Figs/DO_laura_vanderploeg.jpg" 
  alt="DO" 
  style="width: 300px; float:right"/>

- 283 mice (generations 4 & 5)
- 320 (of 7851) SNP markers
- phenotype = `OF_immobile_pct` (of 1000s)
- <span style="font-size: 75%">Data: <https://github.com/rqtl/qtl2data/></span>  
- <a href="http://dx.doi.org/10.1007/s00335-014-9508-0"><span style="font-size: 75%">Recla, Robledo, Gatti, Bult, Churchill, Chesler (2014)</span></a>

## Attie/Jax DO population

- 8 CC founder strains (generation 19-22)
- 500 mice in 5 waves
- multiple traits measured
    + 150K SNP GIGA-MUGA chip imputed to 40M SNPs
    + 100s clinical traits (insulin secretion)
    + 30K RNA-Seq expression traits
    + 2K proteomic, 200 metabolomic, 200 lipidomic
    + microbiome: 2K of 16s; 1M of sequencing



