---
title: "StatisticalModels"
author: "Brian S. Yandell, UW-Madison"
date: "January 2017"
output:
  ioslides_presentation: default
  beamer_presentation: default
  slidy_presentation: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Models: relate phenotype to genotype

how does genotype influence phenotype?

Goals for genetic architecture

- Identify quantitative trait loci (QTL)
    + (and interactions among QTL)
- Interval estimates of QTL location
- Estimated QTL effects

Goals for genome-wide selection

- Predict breeding value of individuals
- Select best individuals

## Evolution of QTL models

- original ideas developed when markers were rare & costly
- refinements as technology advanced

- single marker regressions
- single locus models: interval mapping for QTL
- association mapping (including GWAS)
- account for non-locus genetic effects

## Phenotype data 

![](../Fig/Bnapus_pheno.jpg)

<span style="font-size: 75%">Satagopan JM, Yandell BS, Newton MA, Osborn TC (1996) Genetics</span>

## Genotype data

Genetic map

![](../Fig/Bnapus_map.jpg)

## Genotypes on chr N2

![](../Fig/Bnapus_genotype.jpg)

## Genotypes reordered by `flower4`

![](../Fig/Bnapus_genotype4.jpg)

## Marker regression (BC or DH)

<img src="../Fig/Bnapus_anova.jpg" 
  alt="anova" 
  style="width: 400px; float:right"/>

- Also known as ANOVA
- Split sample into groups
    + by genotype at marker
    + red = missing genotype
- Do a t-test or ANOVA
- Repeat for each marker

## Marker regression model

$$ y = \mu_m + e = \mu + m\beta + e $$

- $y$ = phenotypic trait
- $m$ = marker genotype (0,1)
- $\mu$ = reference mean (when $m$=0)
- $\beta$ = marker (allele substition) effect
- $e$ = error = unexplained variation

Marker regression:

- fit model for each marker across genome
- pick most significant marker

## Pros & cons of marker regression

Advantages

- Simple
- Easily incorporates covariates
- Easily extended to more complex models
- No need for genetic map
- ignores marker position on genome

Disadvantages

- Excludse individuals with missing genotype data
- Imperfect information about QTL location
- Suffers in low density scans
- Only considers one QTL at a time

## Interval mapping (IM)

Statistical structure

![](../Fig/structure_bw.pdf)

- missing data problem:  Markers $\longleftrightarrow$ QTL
- model selection problem: QTL, covariates $\longrightarrow$ phenotype

## IM: Lander & Botstein (1989)

- Assume a single QTL model.
- posit each genome position $\lambda$, one at a time, as putative QTL
    + $q$ = genotypes at locus $\lambda$

$$ pr(y | q): y = \mu_q + e = \mu + q\beta + e$$

- mixing proportions over flanking markers
$$ pr(q | m): \texttt{table of proportions}$$

- model is mixture over possible QTL genotypes $X$
- mixture of normals

## Genotype probabilities

<img src="../Fig/genoprob1_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Fig/genoprob2_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Fig/genoprob3_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Fig/genoprob4_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Fig/genoprob5_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Fig/genoprob6_bw.pdf"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Normal mixtures

<img src="../Fig/mixtures_bw.pdf" 
  alt="mix" 
  style="width: 350px; float:right"/>

<img src="../Fig/mixchr.pdf" style="width: 300px"/>

- 2 markers separated by 20 cM
    + QTL closer to left marker
- phenotype distribution
    + given marker genotypes
- mixture components
    + dashed curves

## Interval mapping details

QTL given markers: $p_{ij} = \Pr(q_i = j | m), i=1,\cdots,n;j=0,1$ 

phenotype given QTL: $y_i | q_i \sim N(\mu_{q_i},\sigma^2), i=1,\cdots,n$

 $$\Pr(y_i | m,\mu_0,\mu_1,\sigma) =
\sum_j p_{ij} \, f(y_i; \mu_j,\sigma)$$

normal density: $f(y; \mu,\sigma)= \exp[-(y-\mu)^2/(2\sigma^2)] / \sqrt{2 \pi\sigma^2}$

Log likelihood:

$$l(\mu_0,\mu_1,\sigma) = \sum_i \log \Pr(y_i | m,\mu_0,\mu_1,\sigma)$$

find $\hat{\mu}_0$, $\hat{\mu}_1$, $\hat{\sigma}$ to maximize $l(\mu_0,\mu_1,\sigma)$ (MLEs) 

## EM algorithm (Dempster et al.\ 1977)

**E step:** (pseudo)weights 
$$w_{ij}^{(k)} = \Pr(q_i = j | y_i,m,\hat{\mu},\hat{\sigma})
 = \frac{p_{ij} \,
f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}{
\sum_j p_{ij} \, f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}$$

**M step:** (pseudo)values: 

$$\hat{\mu}_j^{(k)} = \sum_i y_i w_{ij}^{(k)} / \sum_i
w_{ij}^{(k)}$$

$$\hat{\sigma}^{(k)} = \sqrt{ \sum_i \sum_j w_{ij}^{(k)}
(y_i-\hat{\mu}_j^{(k)})^2/n}$$

**EM algorithm:** Set $w_{ij}^{(1)} = p_{ij}$; iterate E\&M steps to
convergence

## Pros and Cons of IM

Pros:

- handles missing data
- locus may be apart from marker

Cons:

- still only one locus affecting trait

## Multiple loci models

$$ y = \mu_q + e $$

$$ \mu_q = \mu(q_1, q_2, \cdots) = \mu+q_1\beta_1+q_2\beta_2+\cdots$$

- allows for multiple loci
- can add epistasis

$$ \mu_q  = \mu+q_1\beta_1+q_2\beta_2+q_{12}\gamma $$

## Shortcoming of multiple loci models

- only inlcudes significant loci
- but only include significant loci in practice
- selection bias: 
    + overestimate effects of included loci
    + many loci of small effect ignored ...

## Missing heritability (Eleazar Eskin)

Rewrite QTL genotype $q_j$ as $X_j$ with mean 0, variance 1 (depends on frequency of the genotype in population).

Only pick up significant QTL


$$ \mu_q  = \mu+\sum_{j=1}^k q_j\beta_j $$

but actual model has $K>k$ terms

$$ \mu_q  = \mu+\sum_{j=1}^K q_j\beta_j $$
