---
title: "StatisticalModels"
author: "Brian S. Yandell, UW-Madison"
date: "January 2017"
output:
  ioslides_presentation: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Models: relate phenotype to genotype

how does genotype influence phenotype?

Goals for genetic architecture

- Identify quantitative trait loci (QTL)
    + (and interactions among QTL)
- Interval estimates of QTL location
- Estimated QTL effects

Goals for genome-wide selection

- Predict breeding value of individuals
- Select best individuals

## PHE = GEN + ENV

GEN = QTL + GWA

- QTL ($\mu_q$) is local to genomic region
- GWA depends on population structure ($K$ = kinship)
    + measures relationships away from QTL
    + GWA = $Zu$ = $K'Ku$

ENV = design + predictors + error

- design structure (blocks, locations)
- predictor structure (soil additives)
- measurement error
    
<span style="font-size: 75%">Falconer & Mackay (1960--1996)</span>

## Strategy for mapping

- Want to figure out what is going on
    + preliminary search: find important story
    + need strategies to uncover patterns
- Want to tell story in publication

- How to accomplish mapping goal
    + organic search for patterns
    + organize methods as you go
    + document steps (so you can redo)

## Challenges of models for experiments 

Simpler models yield clearer results

- compare 2 conditions 
- examine linear trend
- control for other factors

But reality may be more complicated

- masking of genetic effect (by background, etc.)
- subtle timing (when to measure)
- hard to measure key features (shape, quality)
- unknown details of processes under study

## Evolution of QTL models

- original ideas developed when markers were rare & costly
- models & methods refined over time as technology advanced

- single marker regression
- QTL (quantitative trait loci)
    + single locus models: interval mapping for QTL
    + multiple QTL models: epistasis
- GWA (genome-wide association mapping)
    + adjust for population structure
    + capture "missing heritability"
- multi-parent populations (MPP)

## Continuum of QTL to GWA

<img src="../Figs/pareto.jpg" alt="Pareto Diagram" style="width: 800px"/>

## Why study multiple traits?

- We may not know ...
    + how to measure (quality, shape, taste)
    + when to measure (development, season, kinetics)
    + what to measure (molecular pathways)
- We are interested in many things
    + individuals expensive
    + multiple measurements cheap (high throughput)
- There may be pleiotropy vs close linkage
    + correlated through one cause or linkage drag?
    + untangle genetics and environment

## Multiple traits and GxE

- same phenotype in multiple environments
    + does genetic response depend on environment?
    + comparing 2 (or more) environments
    + trends across a cline or gradient
- multiple phenotypes
    + does one trait affect others (directly)?
    + leveraging similar biological function
    + correlation and causalty models

## Phenotype data 

![](../Figs/Bnapus_pheno.jpg)

<span style="font-size: 75%">Satagopan JM, Yandell BS, Newton MA, Osborn TC (1996) Genetics</span>

## Genotype data

Genetic map

![](../Figs/Bnapus_map.jpg)

## Genotypes on chr N2

![](../Figs/Bnapus_genotype.jpg)

## Genotypes reordered by `flower4`

![](../Figs/Bnapus_genotype4.jpg)

## Marker regression (BC or DH)

<img src="../Figs/Bnapus_anova.jpg" 
  alt="anova" 
  style="width: 400px; float:right"/>

- Also known as ANOVA
- Split sample into groups
    + by genotype at marker
    + red = missing genotype
- Do a t-test or ANOVA
- Repeat for each marker

## Marker regression model

$$ y = \mu_m + e = \mu + m\beta + e $$

- $y$ = phenotypic trait
- $m$ = marker genotype (0,1)
- $\mu$ = reference mean (when $m$=0)
- $\beta$ = marker (allele substition) effect
- $e$ = error = unexplained variation

Marker regression:

- fit model for each marker across genome
- pick most significant marker

## Pros & cons of marker regression

Advantages

- Simple
- Easily incorporates covariates
- Easily extended to more complex models
- No need for genetic map
- ignores marker position on genome

Disadvantages

- Excludse individuals with missing genotype data
- Imperfect information about QTL location
- Suffers in low density scans
- Only considers one QTL at a time

## Statistical structure

![](../Figs/structure_bw.jpg)

- missing data problem:  Markers $\longleftrightarrow$ QTL
- model selection problem: QTL, covariates $\longrightarrow$ phenotype

## Interval mapping (IM)

- Assume a single QTL model.
- posit each genome position $\lambda$, one at a time, as putative QTL
    + $q$ = genotypes at locus $\lambda$

$$ pr(y | q): y = \mu_q + e = \mu + q\beta + e$$

- mixing proportions over flanking markers
$$ pr(q | m): \texttt{table of proportions}$$

- model is mixture over possible QTL genotypes $X$
- mixture of normals

<span style="font-size: 75%">Lander & Botstein (1989) Genetics</span>

## Genotype probabilities

<img src="../Figs/genoprob1_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Figs/genoprob2_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Figs/genoprob3_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Figs/genoprob4_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Figs/genoprob5_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Genotype probabilities

<img src="../Figs/genoprob6_bw.jpg"
  style="width: 800px"/>

Calculate $pr(q|m)$ assuming

- No crossover interference
- No genotyping errors

Or use the hidden Markov model (HMM)} technology

- To allow for genotyping errors
- To incorporate dominant markers

## Phenotype as mixture across unknown genotypes

<img src="../Figs/mixtures_bw.jpg" 
  alt="mix" 
  style="width: 350px; float:right"/>

<img src="../Figs/mixchr.jpg" style="width: 300px"/>

- 2 markers separated by 20 cM
    + QTL closer to left marker
- phenotype distribution
    + given marker genotypes
- mixture components
    + dashed curves

## [Interval mapping idea](http://www.biostat.wisc.edu/~kbroman/D3/em_alg/)

think marker regression with fuzzy groups

<img src="../Figs/em_alg_illustration.png" alt="Interactive EM illustration" style="width: 500px; float:left"/>

<img src="../Figs/Bnapus_anova.jpg" 
  alt="anova" 
  style="width: 250px; float:right"/>



## Interval mapping details

QTL given markers: $p_{ij} = \Pr(q_i = j | m), i=1,\cdots,n;j=0,1$ 

phenotype given QTL: $y_i | q_i \sim N(\mu_{q_i},\sigma^2), i=1,\cdots,n$

 $$\Pr(y_i | m,\mu_0,\mu_1,\sigma) =
\sum_j p_{ij} \, f(y_i; \mu_j,\sigma)$$

normal density: $f(y; \mu,\sigma)= \exp[-(y-\mu)^2/(2\sigma^2)] / \sqrt{2 \pi\sigma^2}$

Log likelihood:

$$l(\mu_0,\mu_1,\sigma) = \sum_i \log \Pr(y_i | m,\mu_0,\mu_1,\sigma)$$

find $\hat{\mu}_0$, $\hat{\mu}_1$, $\hat{\sigma}$ to maximize $l(\mu_0,\mu_1,\sigma)$ (MLEs) 

## EM algorithm (Dempster et al.\ 1977)

**E step:** (pseudo)weights 
$$w_{ij}^{(k)} = \Pr(q_i = j | y_i,m,\hat{\mu},\hat{\sigma})
 = \frac{p_{ij} \,
f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}{
\sum_j p_{ij} \, f(y_i; \hat{\mu}_j^{(k-1)},\hat{\sigma}^{(k-1)})}$$

**M step:** (pseudo)values: 

$$\hat{\mu}_j^{(k)} = \sum_i y_i w_{ij}^{(k)} / \sum_i
w_{ij}^{(k)}$$

$$\hat{\sigma}^{(k)} = \sqrt{ \sum_i \sum_j w_{ij}^{(k)}
(y_i-\hat{\mu}_j^{(k)})^2/n}$$

**EM algorithm:** Set $w_{ij}^{(1)} = p_{ij}$; iterate E\&M steps to
convergence

## LOD Scores

LOD score measures strength of evidence for QTL at a locus $\lambda$ as 
$\log_{10}$ likelihood ratio of model with QTL at $\lambda$ to no QTL model

$\texttt{lod}(\lambda) = [l(\hat{\mu}_{0\lambda}, \hat{\mu}_{1\lambda}, \hat{\sigma}_\lambda) - l(\hat{\mu}, \hat{\sigma})]/\log(10)$

$\hat{\mu}_{0\lambda}, \hat{\mu}_{1\lambda}, \hat{\sigma}_\lambda$ are MLEs with QTL at $\lambda$

No QTL model: $y$ has mean $\mu$, variance $\sigma^2$

## Pros and Cons of IM

Pros:

- handles missing data
- locus may be apart from marker

Cons:

- still only one locus affecting trait

## Multiple loci models

$$ y = \mu_q + e $$

$$ \mu_q = \mu(q_1, q_2, \cdots) = \mu+q_1\beta_1+q_2\beta_2+\cdots$$

- allows for multiple loci
- can add epistasis

$$ \mu_q  = \mu+q_1\beta_1+q_2\beta_2+q_{12}\gamma $$

## Shortcoming of multiple loci models

- only inlcudes significant loci
- but only include significant loci in practice
- selection bias: 
    + overestimate effects of included loci
    + many loci of small effect ignored ...

## GWA big idea

- Kinship via pedigree: all we had
    + average / predicted relationship
    + works globally, might be inaccurate locally
    + think maternal vs paternal twins
- Kinship via SNP/GBS
    + estimated from data
    + used to be selected to be neutral
    + now use all but away from QTL
- GWA measures ...
    + population structure
    + other genetic effects including other QTL

## Missing heritability (Eleazar Eskin)

Rewrite QTL genotype $q_j$ as $X_j$ with mean 0, variance 1 (depends on frequency of the genotype in population).

Only pick up significant QTL


$$ \mu_q  = \mu+\sum_{j=1}^k q_j\beta_j $$

but actual model has $K>k$ terms

$$ \mu_q  = \mu+\sum_{j=1}^K q_j\beta_j $$
