---
title: "Multiple Traits"
author: "Brian S. Yandell, UW-Madison"
date: "January 2017"
output:
  ioslides_presentation: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Why study multiple traits?

- We may not know ...
    + how to measure (quality, shape, taste)
    + when to measure (development, season, kinetics)
    + what to measure (molecular pathways)
- We are interested in many things
    + individuals expensive
    + multiple measurements cheap (high throughput)
- causal relationships among traits
    + pleiotropy vs close linkage
    + indirect effect of QTL via other trait
    + untangle genetic & environment correlation

## Multiple traits and GxE

- same phenotype in multiple environments
    + does genetic response depend on environment?
    + comparing 2 (or more) environments
    + trends across a cline or gradient
- multiple phenotypes
    + does one trait affect others (directly)?
    + leveraging similar biological function
    + correlation and causalty models

## What affects outcomes?

- Time
    + <span style="font-size: 75%">development, growth, aging, senescence</span>
    + <span style="font-size: 75%">physiology, reaction to stimuli</span>
    + <span style="font-size: 75%">evolution, selection</span>
- Space
    + <span style="font-size: 75%">geography, altitude, longitude, height, gravity</span>
    + <span style="font-size: 75%">soil structure, plant density</span>
    + <span style="font-size: 75%">seed & pollen (plant part) dispersal</span>
- Stress
    + <span style="font-size: 75%">physiology, nutrients, light, heat, water</span>
    + <span style="font-size: 75%">weeds, herbivores, pollinators</span>
    + <span style="font-size: 75%">disease, pests, parasites, harvesting</span>

## Goals

- use multiple phenotypes to improve QTL detection
- do traits share QTL (pleiotropy)?
- causal relationships among traits?
- effects of QTL across time

## gravitopic response: root tip

<img
  src="../Figs/genetics.113.153346.gif" alt="root tip"
  style="width: 350px; float:right"/>

- _Arabidopsis thaliana_ Ler x Cvi
    + 92 NIL; 2525 seedlings
    + 162 RILs; 2132 (RIL1) or 2325 (RIL2) seedlings
- genotypes: 102 (NIL) or 234 (RILs) markers on 5 chr

- root tip angles every 2 min

<span style="font-size: 75%">Moore, Johnson, Kwak, Livny, Broman, Spalding (2013)</span>
  
## [gravitropic response over time](http://kbroman.org/qtlcharts/example/iplotMScanone.html)

<a href="http://kbroman.org/qtlcharts/example/iplotMScanone.html"><img
  src="../Figs/Screen Shot_iplotMScanone.png" alt="Interactive scan over time"
  style="width: 600px"/></a>

<span style="font-size: 75%">Moore, Johnson, Kwak, Livny, Broman, Spalding (2013)</span>

## Pleiotropy?

<img src="../Figs/pleiotropy_network_bw.jpg" alt="pleiotropy" style="width: 400px; float:right"/>

- $y_1$ & $y_2$ are pleiotropic
    + both depend on $q_1$
    + cor($y_1,y_2|q_1$) = 0
- $y_2$ & $y_3$ are pleiotropic
    + both depend on $q_2$
    + cor($y_2,y_3|q_2$) = 0
- $y_1$ & $y_3$ have linked QTLs
    + cor($q_1,q_2$) via linkage
    + cor($y_1,y_3$) indirect 

## Close linkage?

<img src="../Figs/causal_network_bw.jpg" alt="close linkage" style="width: 800px"/>

## Basic model

one trait:
$$ y = q \beta + e, \qquad e \sim \text{N}(0, \sigma^2) $$

$$ \text{LOD} = (n/2) \log_{10} ( \text{RSS}_0 /
  \text{RSS}_1 ) $$

multiple traits:

$$ Y = Q \boldsymbol{\beta} + E,
  \qquad E \sim \text{MVN}(0, \Sigma) $$

$$ \text{LOD} = (n/2) \log_{10} ( {|\hat{\Sigma}_0|} /
  { |\hat{\Sigma}_1|}) $$
  
## Pleiotropy?

- consider two traits and one chromosome
- assume each affected by a single QTL
- 1-D scan for single QTL assumed to affect both
- 2-D scan over chromosome
    + separate axes for each trait
    + or profile each trait across other trait
- significance?
    + parametric bootstrap using fitted single-QTL model
    + stratified (within QTL genotype) permutation test

## Pleiotropy in flowering time

![](../Figs/Bnapus_1qtl.jpg)

## Pleiotropy in flowering time

<img src="../Figs/Bnapus_linkage.jpg" alt="pleiotropy flower" style="float:right"/>

- allow 2 QTL per trait  
- is 1 enough?
- or are there 2 QTL?

## Pleiotropy in flowering time

profile LOD of each trait with respect to the other

![](../Figs/Bnapus_pleiotropy.jpg)

## causal models

- All ... phenomena are linked together, and the problem ... is how close is the degree of association. Karl Pearson (1911)
- direct influence of one condition on another ... [through] all connecting paths of influence ... among the variables in a system with ... causal relations. Sewall Wright (1921)
- Causality is not mystical or metaphysical. It can be understood in terms of simple processes ... in a friendly mathematical language, ready for computer analysis. Judea Pearl (2000)

## causal relations

<img src="../Figs/causal_network_bw.jpg" alt="close linkage" style="width: 500px; float:right"/>

- $q_j$ = genotype (QTL) 
- $r_k$ = RNA expression
- $y$ = physiology

QTLs: <span style="font-size: 75%">$r_j = \mu_{qj} + e_j$</span>  
linkage: <span style="font-size: 75%">$\texttt{cor}(q_1,q_2)$</span>  
causal: <span style="font-size: 75%">$y = r_2 + e_2$</span>   
reactive: <span style="font-size: 75%">$r_4 = a + by + e_4$</span>   

<span style="font-size: 75%">$$\texttt{pr}(Q,R,Y)=
\texttt{pr}(q_1,q_2) 
\texttt{pr}(y_1|q_1)  
\texttt{pr}(r_2|q_2)  
\texttt{pr}(r_3|q_2)  
\texttt{pr}(y|r_2) 
\texttt{pr}(r_4|y)$$</span>

## Nfatc2 impact on LOD

<img
  src="../Figs/nfat_plosgen.jpg" alt="nfat impact on LOD"
  style="width: 400px; float:right"/>

- 500 mice F2: B6 x BTBR obese
- diabetes & obesity
- insulin resistance
- 130+ mRNA expression hot spot
- Nfatc2 as key driver

<a href="http://dx.doi.org/10.1371/journal.pgen.1006466"><span style="font-size: 75%">Keller et al. Yandell, Kendziorski, Attie (2016) PLoS Gen</span></a>

## Nfatc2 causal archicture

is gene mRNA causal (red) or reactive (blue) to other mRNA?

<img
  src="../Figs/nfat_causal_arch.png" alt="nfat causal architecture"
  style="width: 600px"/>

