---
title: "Genome-Wide Association"
author: "Brian S. Yandell, UW-Madison"
date: "January 2017"
output:
  ioslides_presentation: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## evolution of QTL models

original ideas focused on rare & costly markers  
models & methods refined as technology advanced

- single marker regression
- QTL (quantitative trait loci)
    + single locus models: interval mapping for QTL
    + QTL model search: QTLs & epistasis
- **GWA (genome-wide association mapping)**
    + adjust for population structure
    + capture "missing heritability"
    + genome-wide selection

## GWA big idea

- only detect some genetic effects
    + significant QTL
- effects of modest or small effect ignored
    + non-significant QTL
    + effects too small to observe or test
- these other effects have two sources
    + many small effects on phenotype
    + population admixture reflected in genome structure

## missing heritability

"actual" genetic model $y = \mu_q + e$  
with $J$ genetic effects (recode $q_j$ to have variance 1)

$$\mu_q  = \mu+\sum_{j=1}^J \beta_j q_j$$

actual heritability: 
$$h^2 = \frac{\sum_{j=1}^J \beta_j^2}{\sigma^2 + \sum_{j=1}^J \beta_j^2}$$
(consider backcross and ignore epistasis from here forward)

## best QTL misses most of heritability

but "best" QTL model has $2$ terms

$$ \mu_q  = \mu+ \beta_1 q_1 + \beta_2 q_2$$
with heritability:
$$h_\texttt{QTL}^2 = \frac{\beta_1^2 + \beta_2^2}{\sigma^2 + \sum_{j=1}^J \beta_j^2}$$

missing genetic variability: 
$$h^2 - h_\texttt{QTL}^2 > 0$$

<span style="font-size: 75%">Eskin (2105) <http://dx.doi.org/10.1145/2817827></span>

## population structure inflates effects

- spurious association of phenotype
- population structure affects
    + some phenotypes
    + some genetic loci
- but genotype may not affect phenotype
    + if we adjust for population structure

## mouse strains & body weight

<img src="../Figs/eskin-population-structure-draft.jpg" alt="Eskin population structure" style="width: 700px"/>

<span style="font-size: 50%">Eskin (2106 draft) Review doi:10.1101/092106</span>

## indirect correlation

<img src="../Figs/Eskin_indirect.png" alt="Eskin indirect" style="width: 700px"/>

<span style="font-size: 50%">Eskin (2106 draft) Review doi:10.1101/092106</span>

## mixed model: QTL + GWA

$$y = \mu_q + g + e$$

$\mu_q$ = QTL effects (fixed)  
$g$ = GWA effects (random)  
$$g\sim N(0,\sigma_g^2 K)$$  
e = unexplained variation (random)  
$$e\sim N(0,\sigma^2 I)$$

$K$ = kinship matrix  
$I$ = identity matrix (1s on diagonal, 0s off diagonal)

## GWA and kinship $K$

- estimate kinship $K$ via pedigree: all we had in past
    + average / predicted relationship
    + works globally, might be inaccurate locally
    + think siblings vs parent/offspring
- estimate kinship $K$ via SNP or GBS
    + estimate $K$ from marker data $M$
    + used to select markers to be neutral
    + now use markers away from QTL $q$

$$K = c*M^{\text{T}}M$$
$c$ set so diagonal of $K$ is 1

## PHE = QTL + GWA + ENV example

![](../Figs/phe_qtl_gwa.jpg)

## fitting the mixed model

distribution of phenotype
$$y = \mu_q + g + e$$
$$y \sim N(\mu_q,V), V =\sigma_g^2K + \sigma^2I$$

iterate to solve (similar to EM idea)

- get MLE of $\mu_q$ given $V$: $\hat{\mu}_q=(V^{\text{T}}V)^{-1}V^{\text{T}}y$
- estimate $\sigma_g$ and $\sigma^2$ given $\mu_q$

## mixed model software for QTL and GWA

[EMMA (Efficient Mixed Model Association)](http://mouse.cs.ucla.edu/emma/)

- Convert matrix operations to vector operations
    + with one clever eigendecomposition
    + 1000x faster than TASSEL/SAS implementation
    + O(n) per iteration
    + accurate and reliable
- Eleazar Eskin (UCLA) standalone package

## EMMA software spinoffs

[rrBLUP](http://pbgworks.org/node/1440)

- EMMA adapted to genotypic prediction
    + `A.mat` to compute $K$ from markers
    + `kin.blup` to predict phenotypes
- <span style="font-size: 75%">Jeff Endelman (UW-Madison) R package</span>

[qtl2scan](http://kbroman.org/qtl2/)

- EMMA adapted to R/qtl2 package
- leave out one chromosome method
    + `calc_kinship` with `type="loco"` to compute $K$
    + `scan1` with `kinship` option to fit mixed model
- <span style="font-size: 75%">Karl Broman (UW-Madison) R package suite</span>

## Diversity Outbred experiment

<img src="../Figs/DO_laura_vanderploeg.jpg" 
  alt="DO" 
  style="width: 300px; float:right"/>

- 283 mice 
- Diversity Outbred cross
- generations 4 & 5
- 320 (of 7851) SNP markers
- phenotype = `OF_immobile_pct`
- <span style="font-size: 75%">Data: <https://github.com/rqtl/qtl2data/></span>  
- <a href="http://dx.doi.org/10.1007/s00335-014-9508-0"><span style="font-size: 75%">Recla, Robledo, Gatti, Bult, Churchill, Chesler (2014)</span></a>

## genome scans with kinship

![](../Figs/DO_lod_kinship.jpg)

## detail for key chromosome

![](../Figs/DO_lod_kinship2.jpg)

## allele scan on chr 2

![](../Figs/DO_allele.jpg)

## allele scan on chr 2 with BLUP

![](../Figs/DO_allele_blup.jpg)

## Detailed look in region of peak

- consider SNPs within region of peak
- $t$-test for differences
- SNPs have different patterns across founders
- use SNPs to refine search

## allele vs SNP scans

- allele-based genome scan: LOD maps
    + continuous curve across loci
    + interval mapping for missing data
    + model effect of founder alleles
- DO founder alleles: A,B,C,D,E,F,G,H
- response ~ sum of effects of alleles
- additive model
    + 1-QTL allele-based genotype $q$ = {$a_1,a_2$}
    + $\mu_q = \mu + a_1 + a_2$
    + test if all $a$s are the same
    + 8 unknown parameters

## SNP-based genome scan

- SNP-based genome scan: GWAS Manhattan plots
    + discrete tests of SNPs or other features
    + typically 2 SNP alleles (1 is reference)
    + model effect of number of non-ref SNP copies 
- SNP recorded as pair of DNA base pairs (A,C,G,T)
    + SNPs typically have two values (G/T)
    + individual has genotype GG, GT or TT

## SNP scans

![](../Figs/DO_SNP.jpg)
    
## SNP-based genome scan details

- simplifed to number of copies of non-reference allele
    + SNP genotypes $s$ = 0,1
    + DO reference is B = B6
- mixed model $y= a+bs+g+e$
    + fixed part $\mu_q = a+bs$
    + GWA mixed part $g$
    + unexplained variation $e$
- test slope $b=0$ using LOD score

## SNP patterns

![](../Figs/DO_SNP_pattern.jpg)

## SNP best patterns

![](../Figs/DO_SNP_pattern_best.jpg)

## additive model is quick & easy

- good news
    + fewer parameters to understand
    + easy to build tools
    + fast to run
    + relate LOD to allele effects directly
    + likely good model for mRNA expression
- bad news
    + could miss important gene action information
    + could miss loci (false negatives)
    + could detect false positives
    + LOD and allele effects may conflict due to dominance

## full model (with dominance)

- DO founder alleles: A,B,C,D,E,F,G,H
- response ~ effect of pair of alleles
- full model for allele-based scan
    + $\mu_q = \mu + a_1 + a_2 + d_{12}$ (additive & dominance effects)
    + $E(y) = \texttt{mean}(A_1,A_2)$ (alleles $A_1$, $A_2$)
    + test if all means are the same (all $a$s equal, all $d$s zero)
    + 36 unknown parameters
- full model for SNP-based scan: $s$ = 0,1,2
    + $E(y) = \texttt{mean}(s)$ (3 unknowns)
    + test if all means are the same

## gene/SNP action

- study additive & dominance of single trait
- compare co-mapping traits -- same gene action?
- extend discrete SNP-based scan to continuous
    + scan of particular allele contrasts
    + LOD and allele contrast scans
    
```{r echo=FALSE, fig.height=3}
data <- data.frame(x=rep(0:2,4),y=c(0:2,0,1.5,2,0,2,2,0,0,2),
                   z=rep(c("additive","general","dominant","recessive"), each = 3))
library(ggplot2)
ggplot(data, aes(x,y)) +
  geom_point(size=4) +
  geom_line(size=2) +
  facet_grid(~z) +
  labs(x="",y="phenotype mean") + 
  theme(axis.text.y = element_blank()) +
  scale_x_continuous(breaks=0:2)
```

## what is genomic selection?

use statistical modeling to predict how a plant will perform  
before it is field-tested

- genomic selection (GS)
    + marker assisted selection (MAS)
    + genome-wide selection (GS)
- other uses of word (relevent to systems genetics)
    + natural selection: survival of the fittest
    + model selection: search for QTLs
    + selection bias: overestimate of QTL effects

## why use genomic selection?

- trait is highly polygenic (genetically variable)
    + influenced by a few key genomic regions
    + high heritability (low environmental variation)
- measuring trait is costly
    + difficult or expensive process (technology)
    + measuring tool may be highly variable
    + time-consuming (plant has to grow first)
    + desire to streamline multi-year selection

## what is genomic selection?

- forms of genome-wide selection
    + marker-assisted: with phenotypes
    + marker-based: without phenotypes
    
- use markers to improve selection for complex traits
    + predict phenotype from marker genotype
    + select candidates based on best marker genotypes
    + use training set to predict test set of individuals

## old paradigm: marker prediction

- 1990s & 2000s: markers were expensive
- economic strategy:
    + first identify significant markers (QTL analysis)
    + use best markers to genotype selection candidates
- estimate marker effects by multiple regression
    + treat genetic effects as fixed and few
    + $E(y)= \mu_q, q=(q_1,q_2,q_3)$

## marker assisted selection (MAS)

![](../Figs/pedigree-versus-marker-assisted-breeding.gif)


<span style="font-size: 50%">www.21stcentech.com/heard-marker-assisted-breeding/</span>

## new paradigm: use "all" markers

- new paradigm with technology advances
    + improved statistical methods and software
    + cheap markers
- using only significant markers to predict trait ...
    + gives good estimates (maybe) of markers ...
    + but does not maximize accuracy
- simple but effective approach
    + treat marker effects as random
    + use all markers (away from QTL if any)

## old vs new

<img src="../Figs/MAS_Endelman.png" alt="old vs new GS" style="width: 750px"/>

## mixed model approach

MAS approach $y = \mu_q + e, V(e) = \sigma^2I$

- estimate fixed QTL effects $\hat{\mu}_q$ (MLEs)
- predict phenotype using fixed effects $\hat{y}=\hat{\mu}_q$

GS approach  $y = \mu + g + e, V(g) = \sigma_g^2K$

- estimate kinship $K$ from all markers $M$ as for GWA
- predict random effect $\hat{g}$ using BLUP
- predict phenotype $\hat{y}=\hat{g}$

## GS example


