---
title: "Diversity Outbred Micxe"
author: "Brian S. Yandell"
date: "12/16/2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is a study of Diversity Outbred (DO) mice derived from 8 founder strains.

Data: <https://github.com/rqtl/qtl2data/>  
Reference: Recla JM, Robledo RF, Gatti DM, Bult CJ, Churchill GA, Chesler EJ (2014) Precise genetic mapping and integrative bioinformatics in Diversity Outbred mice reveals Hydin as a novel pain gene. Mamm Genome 25:211-222. [doi:10.1007/s00335-014-9508-0](http://dx.doi.org/10.1007/s00335-014-9508-0)
Here are some useful links:

- <http://rqtl.org> (lots of useful stuff here)
    + A brief tour of R/qtl: [[pdf](http://rqtl.org/tutorials/rqtltour.pdf) | [code](http://rqtl.org/tutorials/rqtltour.R)]
    + A shorter tour of R/qtl: [[pdf](http://rqtl.org/tutorials/rqtltour2.pdf) | [code](http://rqtl.org/tutorials/rqtltour2.R)]
    + [R/qtl sample data formats](http://rqtl.org/sampledata/)
    + [A Guide to QTL Mapping with R/qtl by Karl W. Broman & Åšaunak Sen](http://www.rqtl.org/book/)
- <http://kbroman.org/qtl2/>
    + See Installation Instructions
    + [R/qtl User Guide](http://kbroman.org/qtl2/assets/vignettes/user_guide.html)
    + [R/qtl2 input file format](http://kbroman.org/qtl2/assets/vignettes/input_files.html)
    + [R/qtl2ggplot](https://github.com/byandell/qtl2ggplot): Yandell version of qtl2plot with grammar of graphics plots ([ggplot2](http://ggplot2.org))

Files for DO mice are in the <https://github.com/rqtl/qtl2data/tree/master/DOex>.
You will need [DOex.zip](https://github.com/rqtl/qtl2data/blob/master/DOex/DOex.zip) and [c2_snpinfo.rds](https://github.com/rqtl/qtl2data/blob/master/DOex/c2_snpinfo.rds). This is a small subset for teaching purposes of the larger dataset.

## Setup

First we set up needed packages, or libraries.

```{r message=FALSE}
library(qtl2geno)
library(qtl2scan)
library(qtl2ggplot)
library(rrBLUP)
```

Could read in data over the internet. We do it here with local copy,
which is downloaded if not found.

```{r}
do_file <- "DOex.zip"
snp_file <- "c2_snpinfo.rds"
```

Get files from internet if they do not exist locally.

```{r}
web_address <- file.path("https://raw.githubusercontent.com/rqtl/",
               "qtl2data/master/DOex")
if(!file.exists(do_file)) {
  download.file(file.path(web_address, do_file), do_file, quiet=TRUE)
}
if(!file.exists(snp_file)) {
  download.file(file.path(web_address, snp_file), snp_file, quiet=TRUE)
}
rm(web_address)
```

## Read cross object and subset to chr 2.

```{r}
DOex <- read_cross2(do_file)
```

## Calculate genotype probabilities.

Calculate genotype probabilities. This can take some time and may create a huge file.
We reduce to the 8 allele probabilities, which is smaller.
We also put the physical map into the probs object.
For larger datasets, we save both objects, as they took some time to create.

```{r}
pr <- calc_genoprob(DOex, error_prob=0.002)
```

Convert to allele probabilities.
Switch map to physical map in Mbp.

```{r}
apr <- genoprob_to_alleleprob(pr)
apr$map <- DOex$pmap
```

## QTL Scan on chr 2

```{r}
kinship_all <- calc_kinship(apr, "overall")
```

```{r}
pheno_scan <- scan1(apr, DOex$pheno, kinship_all)
```

```{r message=FALSE}
plot(pheno_scan)
```

## QTL scan with kinship adjustment.

Calculate kinship matrices using the `"loco"` option, which calculates kinship for each chromosome using SNPs from all but that chromosome.

```{r}
data1 <- data.frame(y=DOex$pheno[,1],
                    gid=as.numeric(row.names(DOex$pheno)))
```

```{r}
rr_blup <- kin.blup(data1, K = kinship_all,
                 geno="gid", pheno="y")
```

```{r}
round(cor(rr_blup$pred,data1$y),2)
```

```{r}
library(ggplot2)
dat <- data.frame(x = data1$y,
                  y = rr_blup$pred)
jpeg("../../Figs/rrBLUP_DO_GWA.jpg", width=600, height=600)
ggplot(dat, aes(x,y)) + 
  geom_point() +
  geom_smooth(method="lm",se=FALSE) +
  geom_abline(intercept = 0, slope = 1) +
  ylab("rrBLUP prediction") +
  xlab("phenotype")
dev.off()
```

Now try with QTL as covariate.

```{r}
geno <- apr$probs[[1]][,, rownames(max(pheno_scan))]
data1 <- cbind(data1, geno[, LETTERS[c(1,3:8)]])
dip <- apply(geno, 1, function(x) {
    x <- round(2*x)
    if(any(x == 2)) {
      dip <- rep(LETTERS[1:8][x==2], 2)
    } else {
      dip <- LETTERS[1:8][x == 1]
    }
    paste(dip, collapse = "")
  })
```

```{r}
rr_blup_cov <- kin.blup(data1, K = kinship_all,
                 geno="gid", pheno="y",
                 covariate = LETTERS[c(1,3:8)])
```

```{r}
round(cor(rr_blup_cov$pred,data1$y),2)
```

```{r}
fit <- lm(y~A+C+D+E+F+G+H,data1)
round(cor(rr_blup_cov$g,resid(fit)),2)
```

```{r}
library(ggplot2)
dat <- data.frame(x = resid(fit),
                  y = rr_blup_cov$g,
                  dip = dip)
jpeg("../../Figs/rrBLUP_DO_QTL.jpg", width=600, height=600)
ggplot(dat, aes(x,y,label=dip)) + 
  geom_text(aes(col=dip), size=4) +
  geom_smooth(method="lm",se=FALSE) +
  geom_abline(intercept = 0, slope = 1) +
  xlab("residual from allele fit") +
  ylab("rrBLUP GWA term") +
  theme(legend.position = "none")
dev.off()
```

Want to do on test set and training set. Also need to account for how mean and variance changed